diff -urN "MCP-LTS_Clean\\src\\minecraft/net/minecraft/client/Minecraft.java" "MCP-LTS\\src\\minecraft/net/minecraft/client/Minecraft.java"
--- "MCP-LTS_Clean\\src\\minecraft/net/minecraft/client/Minecraft.java"	2021-11-10 11:11:51.701815400 +0100
+++ "MCP-LTS\\src\\minecraft/net/minecraft/client/Minecraft.java"	2021-11-10 22:09:45.419103800 +0100
@@ -58,6 +58,8 @@
 import org.lwjgl.opengl.GL11;
 import org.lwjgl.opengl.GLContext;
 
+import com.mojontwins.modloader.ModLoader;
+
 public final class Minecraft implements Runnable {
     public PlayerController playerController = new PlayerControllerSP(this);
     private boolean fullscreen = false;
@@ -93,7 +95,7 @@
     private MinecraftApplet mcApplet;
     public SoundManager sndManager;
     public MouseHelper mouseHelper;
-    public File mcDataDir;
+    public static File mcDataDir;
     private String server;
     private TextureWaterFX textureWaterFX;
     private TextureLavaFX textureLavaFX;
@@ -310,7 +312,16 @@
 
         long var23 = System.currentTimeMillis();
         int var28 = 0;
-
+        
+        try {
+        	ModLoader.init ();
+        	ModLoader.registerAllTextureOverrides (this.renderEngine);
+        } catch (Exception e) {
+        	e.printStackTrace();
+        	this.shutdownMinecraftApplet();
+        	return;
+        }
+        	
         try {
             while(this.running) {
                 if (this.theWorld != null) {
diff -urN "MCP-LTS_Clean\\src\\minecraft/net/minecraft/game/level/generator/LevelGenerator.java" "MCP-LTS\\src\\minecraft/net/minecraft/game/level/generator/LevelGenerator.java"
--- "MCP-LTS_Clean\\src\\minecraft/net/minecraft/game/level/generator/LevelGenerator.java"	2021-11-10 11:11:52.811206400 +0100
+++ "MCP-LTS\\src\\minecraft/net/minecraft/game/level/generator/LevelGenerator.java"	2021-11-10 22:49:41.089851400 +0100
@@ -2,6 +2,9 @@
 
 import java.util.ArrayList;
 import java.util.Random;
+
+import com.mojontwins.modloader.ModLoader;
+
 import net.minecraft.game.block.Block;
 import net.minecraft.game.block.BlockFlower;
 import net.minecraft.game.level.Spawner;
@@ -405,8 +408,12 @@
         this.guiLoading.displayLoadingString("Building..");
         this.loadingBar();
         this.setNextPhase(0.0F);
+        
         var6.findSpawn();
         generateHouse(var6);
+        
+        ModLoader.generateStructures (this, var6);
+        
         this.guiLoading.displayLoadingString("Planting..");
         this.loadingBar();
         if (this.levelType != 1) {
